amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -345,6 +358,30 @@ void initialize() {
         }
     }
 
+    @VisibleForTesting
+    boolean waitAndCheckIfLeaseTableIsReady() throws InterruptedException,
+            DependencyException, ProvisionedThroughputException, InvalidStateException {
+        long waitTime = ThreadLocalRandom.current().nextLong(MIN_WAIT_TIME_FOR_LEASE_TABLE_CHECK_MILLIS, MAX_WAIT_TIME_FOR_LEASE_TABLE_CHECK_MILLIS);
+        long waitUntil = System.currentTimeMillis() + waitTime;
+
+        while (System.currentTimeMillis() < waitUntil && leaseRefresher.isLeaseTableEmpty()) {
+            // check every 3 seconds if lease table is still empty,
+            // to minimize contention between all workers bootstrapping at the same time
+            log.info("Lease table is still empty. Checking again in {} ms", LEASE_TABLE_CHECK_FREQUENCY_MILLIS);
+            Thread.sleep(LEASE_TABLE_CHECK_FREQUENCY_MILLIS);
+        }
+        return !leaseRefresher.isLeaseTableEmpty();
```

Comment:
We can avoid this scan, if you maintain a flag in loop.
```
boolean isLeaseTableEmpty = false;
while (System.currentTimeMillis() < waitUntil && ( isLeaseTableEmpty = leaseRefresher.isLeaseTableEmpty())) {
...
}
return isLeaseTableEmpty;
```

Reply:
Addressed